src/unix/solarish/mod.rs

```diff
@@ -536,7 +536,7 @@ cfg_if! {
                     .field("ut_session", &self.ut_session)
                     .field("ut_pad", &self.ut_pad)
                     .field("ut_syslen", &self.ut_syslen)
-                    .field("ut_host", &self.ut_host)
+                // .field("ut_host", &self.ut_host)
```

Comment:
The solution is to just coerce the array as a slice, and debug print that instead.

Reply:
I was just following the pattern in linux-gnu utmpx

Reply:
Also when I try to coerce to slice it complains about not knowing that size at compilation time

Reply:
I'm not sure I follow. What error do you get when you write `&self.ut_host as &[_]` ?

Reply:
error E0277:
```sh
error[E0277]: the size for values of type `[_]` cannot be known at compilation time
   --> /home/grayjack/MySources/RustProjects/libc/src/unix/solarish/mod.rs:539:39
    |
539 |                     .field("ut_host", &self.ut_host as &[_])
    |                                       ^^^^^^^^^^^^^^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: the trait `core::marker::Sized` is not implemented for `[_]`
    = note: to learn more, visit <https://doc.rust-lang.org/book/ch19-04-advanced-types.html#dynamically-sized-types-and-the-sized-trait>
    = note: required for the cast to the object type `dyn core::fmt::Debug`
```

Reply:
I think you need to split that up:

```rust
let s = &self.ut_host as &[_];
...
.field("ut_host", s);
```

Reply:
Wrong. Can you try: `.field("ut_host", &&self.ut_host)` ?

Reply:
Well, this give me a huge error on target `x86_64-sun-solaris`, but I tested on the host field of target linux-gnu and worked fine

The error:
```sh
Checking libc v0.2.66 (/home/grayjack/MySources/RustProjects/libc)
error: type does not implement `fmt::Debug`; consider adding `#[derive(Debug)]` or a manual implementation
   --> /home/grayjack/MySources/RustProjects/libc/src/macros.rs:101:21
    |
101 |                       pub union $i { $($field)* }
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /home/grayjack/MySources/RustProjects/libc/src/unix/solarish/mod.rs:406:1
    |
406 | / s_no_extra_traits! {
407 | |     #[cfg_attr(any(target_arch = "x86", target_arch = "x86_64"), repr(packed))]
408 | |     pub struct epoll_event {
409 | |         pub events: u32,
...   |
499 | |     }
500 | | }
    | |_- in this macro invocation
    |
note: lint level defined here
   --> /home/grayjack/MySources/RustProjects/libc/src/lib.rs:32:44
    |
32  | #![cfg_attr(feature = "extra_traits", deny(missing_debug_implementations))]
    |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type does not implement `fmt::Debug`; consider adding `#[derive(Debug)]` or a manual implementation
   --> /home/grayjack/MySources/RustProjects/libc/src/macros.rs:115:13
    |
115 |               pub struct $i { $($field)* }
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /home/grayjack/MySources/RustProjects/libc/src/unix/solarish/mod.rs:406:1
    |
406 | / s_no_extra_traits! {
407 | |     #[cfg_attr(any(target_arch = "x86", target_arch = "x86_64"), repr(packed))]
408 | |     pub struct epoll_event {
409 | |         pub events: u32,
...   |
499 | |     }
500 | | }
    | |_- in this macro invocation
error: type does not implement `fmt::Debug`; consider adding `#[derive(Debug)]` or a manual implementation
   --> /home/grayjack/MySources/RustProjects/libc/src/macros.rs:115:13
    |
115 |               pub struct $i { $($field)* }
    |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
   ::: /home/grayjack/MySources/RustProjects/libc/src/unix/solarish/mod.rs:406:1
    |
406 | / s_no_extra_traits! {
407 | |     #[cfg_attr(any(target_arch = "x86", target_arch = "x86_64"), repr(packed))]
408 | |     pub struct epoll_event {
409 | |         pub events: u32,
...   |
499 | |     }
500 | | }
    | |_- in this macro invocation
```

Reply:
With further investigation, looks like we have 2 structs and 1 union inside `s_no_extra_traits!` invocation, don't implement manually the debug traits

Reply:
If you want to build with the `extra_traits` feature for this target, you need to manually implement the traits for all types inside `s_no_extra_traits!`, or add an `allow(...)` to disable the lint for a particular type.

Reply:
I know, but that's the thing I wasn't me who added these 3 structures inside `s_no_extra_traits!`

Reply:
We only guarantee that `extra_traits` works for the platforms for which it is tested (tier-1 and some tier-2). We do not support `extra_traits` for other platforms, and it is not tested. The only reason is, nobody has added support for it.

If you want to add support for `extra_traits` for these solaris platforms, you'd have to add the impls. If you don't want to add support for `extra_traits` for this platform, then, the simplest thing would be to just don't use the `extra_traits` feature on the platform.

The main reason to add an `allow(...)` is if, for some reason, it is impossible to implement a particular trait for a type, but that's super rare.

Reply:
It's impossible for union types to implement any of the extra traits? What about structs that have union fields?

Reply:
> Is it impossible for union types to implement any of the extra traits?

Not in general, you just have to manually implement them - but for many unions, you don't have enough information to provide a meaningful implementation.

> What about structs that have union fields?

Same, but for structs with union fields you often have more information to provide a meaningful implementation (e.g. a discriminant of some sort somewhere).

Reply:
Thanks, I'll have a look!! :smile: 
I do want solarish to keep supporting extra_traits cause I used it a lot when debugging ('keep' cause it worked till 0.2.65)

Reply:
FWIW I've open an issue to start testing in `ci/build.sh` that `extra_traits` at least builds on all targets in which we currently are able to build libc, but that will take some work, since I expect many targets to be broken.